zellij-server/src/ui/pane_boundaries_frame.rs

```diff
@@ -600,24 +601,26 @@ impl PaneFrame {
             _ => self.empty_title_line(),
         }
     }
-    fn render_title(&self) -> Vec<TerminalCharacter> {
+    fn render_title(&self) -> Result<Vec<TerminalCharacter>> {
         let total_title_length = self.geom.cols.saturating_sub(2); // 2 for the left and right corners
 
         self.render_title_middle(total_title_length)
             .map(|(middle, middle_length)| self.title_line_with_middle(middle, &middle_length))
             .or_else(|| Some(self.title_line_without_middle()))
-            .unwrap()
+            .with_context(|| format!("failed to render title {}", self.title))
```

Comment:
Personally, I like to enclose strings inside print statements into `'` quotes, because the string we print (`self.title`) may be empty, which can be confusing depending on what to print. But that's my personal preference, you can leave it as is, if you prefer!